src/fluent/watch.ts

```diff
@@ -437,6 +492,18 @@ export class Watcher<T extends GenericClass> {
     }
   };
 
+  // Function to handle reconnecting
+  #reconnect = async () => {
+    try {
+      this.#streamCleanup();
+      console.log("Retrying watch");
```

Comment:
console.log looks like code smell here--that's why we emit events generally

Reply:
oops, yeah